src/components/carousel/index.tsx

```diff
@@ -246,6 +246,21 @@ class Carousel extends Component<CarouselProps, CarouselState> {
     }
   };
 
+  getContentOffset = (snapToOffsets: number[] | undefined) => {
+    const {horizontal, initialPage} = this.props;
+    const {initialOffset} = this.state;
+    let contentOffset = initialOffset;
+    if (snapToOffsets && initialPage !== undefined) {
+      const offset = snapToOffsets[initialPage];
+      contentOffset = {
+        x: horizontal ? offset : 0,
+        y: horizontal ? 0 : offset
+      };
+    }
```

Comment:
shouldn't this code be part of what we store in the initialOffset? in the `onContainerLayout` method?

Reply:
Not sure it has to go that way. 
We have a method that calcOffset, we use it for setting the initial offset. 
Now you the logic for initial offset changed a little, Can't we create a method for `calcInitialOffset` that returns the relevant offset and set it in the state? 
What are the blockers here?

Reply:
The new method (with the new logic) `getContentOffset` depends on `snapToOffsets` which is calculated using `containerWidth` which is obtained in `onContainerLayout`.
Technically from testing now, it seems that only the constructor is occurring before this happens, but it feels unsafe to me, WDYT?

Reply:
Ok, I see your point. 
So can we rename the new method to `getInitialContentOffset` so it's clear that it's the initial offset

Reply:
Sure, but I blame RN:
```
/**
 * Used to manually set the starting scroll offset.
 * The default value is {x: 0, y: 0}
 */
contentOffset?: PointPropType | undefined; // zeros
```